enable_rviz: true
sloam:
  two_step_optim: true
  beam_cluster_threshold: 0.5 # max euclidean distance to cluster lidar beams
  default_tree_radius: 0.6 # default tree radius if estimated value seems wrong (due to outliers for example)
  max_dist_to_centroid: 0.6 # consider points further than this parameter to the node centroid to be outliers
  max_ground_dist: 20.0 # max distance from sensor for ground point to be considered
  min_ground_dist: 2.0 # min distance from sensor for ground point to be considered
  max_lidar_dist: 25.0 # max distance from sensor for a point to be considered
  min_landmark_height: 1.0 # if a tree is smaller than this, it will be discarded
  min_landmark_size: 5 # min number of trellis nodes for a tree
  min_vertex_size: 4 # min required number of point in a trellis node
  min_tree_models: 5 # min number of tree models in observation to perform pose estimation
  min_ground_models: 36 # min number of ground models in observation to perform pose estimation
  num_ground_features: 5 # number of points to keep per plane model
  ground_radii_bins: 2 # 5 bins based on the distance from sensor
  ground_theta_bins: 18 # 36 bins relative to angle around lidar
  ground_match_thresh: 1.0 # threshold to match planes
  ground_retain_thresh: 0.05 # keep bottom x% of the points per bin
  features_per_tree: 60 # number of points to keep per tree model
  add_new_tree_thresh_dist: 1.5 # tree must be at least Xm away from other to be added to the semantic map
  rough_tree_match_thresh: 2.5 # rough match before computing the more costly cylinder to cylinder dist
  tree_match_thresh: 0.5 # dist threshold between point and model to use in pose estimation
  odom_freq_filter: 40  # VILENS odometry is 400 hz, lidar is 10 Hz
  min_odom_distance: 0.5
  min_sloam_z: -500.0 # min height of robot for triggering sloam. If you get the error msg "Quad is too low, will not call sloam", reduce this param. 
  map_frame_id: map 
  odom_frame_id: odom_vilens
  robot_frame_id: pandar  # old: base
  # TODO: include sloam/sloam/models/ dir, commit to remote origin unibz-first fork
  seg_model_path: $(env HOME)/Downloads/darknet21pp_hpc.onnx # path to segmentation model
  seg_lidar_fov: 31 # lidar FOV
  seg_lidar_w: 2000 # lidar width
  seg_lidar_h: 32 # lidar height
  cloud_topic: /hesai/pandar
  odom_topic: /vilens/odom
  debug_mode: true
  use_hesai: false
  do_destagger: false
  publish_tf: true
